public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { FC } from 'react';
+import { SelectableValue } from '@grafana/data';
+import { withTypes } from 'react-final-form';
+import {
+  Modal,
+  TextInputField,
+  TextareaInputField,
+  RadioButtonGroupField,
+  validators,
+  LoaderButton,
+} from '@percona/platform-core';
+import { Messages } from './AddStorageLocationModal.messages';
+import { AddStorageLocationFormProps, AddStorageLocationModalProps } from './AddStorageLocationModal.types';
+import { S3Fields } from './S3Fields';
+import { LocalFields } from './LocalFields';
+import { toFormStorageLocation, toStorageLocation } from './AddStorageLocation.utils';
+import { Button, HorizontalGroup } from '@grafana/ui';
+import { LocationType } from '../StorageLocations.types';
+
+const TypeField: FC<{ values: AddStorageLocationFormProps }> = ({ values }) => {
+  const { type, client, server, endpoint, accessKey, secretKey } = values;
+
+  switch (type) {
```

Comment:
I would use a map for this instead of switch.

Reply:
👍